File: Python/1561.py
# 1561. Maximum Number of Coins You Can Get
# https://leetcode.com/problems/maximum-number-of-coins-you-can-get/

class Solution:
  def maxCoins(self, piles: List[int]) -> int:
    piles.sort()
    idx = len(piles) // 3
    out = 0

    while (idx < len(piles)):
      out += piles[idx]
      idx += 2

    return out

File: TypeScript/1561.ts
/**
 * 1561. Maximum Number of Coins You Can Get
 * https://leetcode.com/problems/maximum-number-of-coins-you-can-get/
 */

function maxCoins(piles: number[]): number {
  piles = piles.sort((a: number, b: number) => a - b);
  let idx: number = piles.length / 3;
  let out: number = 0;

  while (idx < piles.length) {
    out += piles[idx];
    idx += 2;
  }

  return out;
}
